docs/design.html

```diff
@@ -216,6 +216,26 @@ <h4><a id="design_offlineload" href="#design_offlineload">Offline Data Load</a><
     In the case of Hadoop we parallelize the data load by splitting the load over individual map tasks, one for each node/topic/partition combination, allowing full parallelism in the loading. Hadoop provides the task
     management, and tasks which fail can restart without danger of duplicate data&mdash;they simply restart from their original position.
 
+    <h4><a id="static_membership" href="#static_membership">Static Membership</a></h4>
+    Static membership aims to improve the availability of stream applications, consumer groups and other applications built on top of the group rebalance protocol.
+    The rebalance protocol relies on the group coordinator to allocate entity ids to group members. These generated ids are ephemeral and will change when members restart and rejoin.
+    For consumer based apps, this "dynamic membership" can cause a large percentage of tasks re-assigned to different instances during administrative operations
+    such as code deploys, configuration updates and periodic restarts. For large state applications, shuffled tasks need a long time to recover their local states before processing
+    and cause applications to be partially or entirely unavailable. Motivated by this observation, new Kafka’s group management protocol allows group members to provide persistent entity ids.
+    Group membership remains unchanged based on those ids, thus no rebalance will be triggered.
+    <p>
+    If you want to use static membership,
+    <ul>
+        <li>Upgrade both broker cluster and stream apps to 2.3 or beyond, and also make sure the upgraded brokers are using <code>inter.broker.protocol.version</code>
+            of 2.3 or beyond as well.</li>
+        <li>Set the config <code>ConsumerConfig#GROUP_INSTANCE_ID_CONFIG</code> to a unique value for each consumer instance under one group.</li>
```

Comment:
Maybe add one bullet point for Streams:
`<li>For Kafka Streams applications, it is sufficient to set a unique <code>ConsumerConfig#GROUP_INSTANCE_ID_CONFIG</code> per `KafkaStreams` instance, independent of the number of used threads for an instance.</li>`

Side question: what happens if one set's the config for a consumer with no group.id -- this case could easily happen for Kafka Streams restore and global consumer?

Reply:
@mjsax Static membership only affects group based consumer app. Standalone consumers are not affected (I.E. we won't assign them `group.instance.id`

Reply:
Let me rephrase: What happens if a user configures no `group.id` but does configure `group.instance.id`. Do the consumer crash? Or is the `group.instance.id` config just ignored?

Reply:
My understanding is that these two configs should not affect each other. But will try to add a unit test to check.